### PR TITLE
Improve app, appCatalogEntry and appCatalog CRD comments

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/14845

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Comment by tomahawk28** [comment]
I will generate files after I get approval.

File: pkg/apis/application/v1alpha1/app_catalog_entry_types.go [hunk 1]
```diff
@@ -47,7 +47,8 @@ func NewAppCatalogEntryCR() *AppCatalogEntry {
 // +kubebuilder:storageversion
 // +k8s:openapi-gen=true
 
-// AppCatalogEntry represents an entry of an app in a catalog of managed apps.
+// AppCatalogEntry represents an entry of an app in a catalog of managed apps. It stores metadata for specific versions and apps.
+// It is generated by app-operator and consumed by app-admission-controller.
```
Comment (tomahawk28) [inline pkg/apis/application/v1alpha1/app_catalog_entry_types.go:51]: No operator will reconcile to appCatalogEntry CRs but `app-admissions-controller`  would use these.

**Review by marians — APPROVED** [review]
Looks like useful clarification! ❤️

**Review by rossf7 — APPROVED** [review]
LGTM 🚀

**Pull request merged**